OneCommunityGlobal/HighestGoodNetworkApp
# yihan eliminate need for additional “save” with adding Projects or Teams

# Description
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/982b3ce2-293b-4095-9c05-a484c86cb8f8)

## Main changes explained:
- Update UserProfile.jsx for (1)adding handleSubmit property on TeamsTab and ProjectsTab, (2) using userProfileRef.current to update user profile, (3) transfering disabled property to TeamsTab and ProjectsTab.
- Update TeamsTab.jsx and Projects.jsx for transfering submit function and disabled property to AddTeamPopup and AddProjectPopup component.
- Update AddTeamPopup.jsx and AddProjectPopup.jsx for executing handleSubmit function when clicking on "confirm" button.
- Update UserProjectsTable.jsx/.css and UserTeamsTable.jsx/.css for disabling assign button and adding toolTip to button when save changes warning exists.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as admin/owner user
4. go to a user's profile → Teams/Projects tab → Assign teams/projects → choose teams/projects → Confirm
5. verify (1) assigned team/project are saved successfully without clicking on "save changes" button, (2) team popup close automatically after click on confirm button, (3) Assign Team and Assign Project button are disabled when save changes warning exist.

## Screenshots or videos of changes:
Assign Team:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/7f83b756-f94c-417d-a3af-b43cf0b82144

Assign Project:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/cb19371b-a246-40b3-875b-e563a3220cfd



## Note:
When assigning projects, you may notice that the save changes warning is still there after clicking on "confirm" button. This bug appears in Dev as well, so I gonna log it in the bug list. As a result, you can refresh the page to check if the assigned projects are saved successfully when testing that part.
